### ISSUE TITLE
BYOL Implementation - loss calculation

### ISSUE THREAD

**Issue by MaveriQ** [PyTorchLightning/lightning-bolts#801]
Hi

In the BYOL [implementation](https://github.com/PyTorchLightning/lightning-bolts/blob/master/pl_bolts/models/self_supervised/byol/byol_module.py) copied below, shouldn't there be a .detach() with z2 in the loss calculations to stop gradient flow in target network?

```py
        # Image 1 to image 2 loss
        y1, z1, h1 = self.online_network(img_1)
        with torch.no_grad():
            y2, z2, h2 = self.target_network(img_2)
        loss_a = -2 * F.cosine_similarity(h1, z2).mean()


        # Image 2 to image 1 loss
        y1, z1, h1 = self.online_network(img_2)
        with torch.no_grad():
            y2, z2, h2 = self.target_network(img_1)
        # L2 normalize
        loss_b = -2 * F.cosine_similarity(h1, z2).mean()
```
For reference, please see [this](https://github.com/lucidrains/byol-pytorch/blob/caa65d7c19a80a3611f4049c638ecd8401f8f3c4/byol_pytorch/byol_pytorch.py#L254-L255) implementation. 

Thanks
